### PR TITLE
Version 3.0

### PR DESCRIPTION
**Pull request by hotline-emu**
- [ ] Update LICENSE file
- [ ] Change package owner to `traderinteractive`
- [ ] Change Namespace to `TraderInteractive`
- [ ] Add .github files  CODEOWNERS, PULL_REQUEST_TEMPLATE, ISSUE_TEMPLATE & CONTRIBUTING
- [ ] Remove support for PHP < 7
- [ ] Switch to PSR2 coding standard
- [ ] Update README
- [ ] Remove composer.lock from version control
- [ ] Update Scrutinizer config
- [ ] Update Coveralls config
- [ ] Update to PHPUnit 6
- [ ] Switch to phpunit.xml.dist config
- [ ] Add parameter and return type hints where possible

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,23 +15,19 @@ util-http-php requires PHP 5.4 (or later).
```
Comment (chadicus) [inline README.md:18]: Now requires php 7.0

File: README.md [hunk 2]
```diff
@@ -15,23 +15,19 @@ util-http-php requires PHP 5.4 (or later).
...
-    "require": {
-        "dominionenterprises/util-http": "~1.0"
-    }
-}
+`traderinteractive/util-http` to your project's `composer.json` file such as:
+
+```sh
+composer require traderinteractive/util-http
 ```
 ##Documentation
```
Comment (chadicus) [inline README.md]: Can you fix these headers?  Add a space between the `##` and the title

File: README.md [hunk 3]
```diff
@@ -15,23 +15,19 @@ util-http-php requires PHP 5.4 (or later).
...
 ##Contact
 Developers may be contacted at:
 
- * [Pull Requests](https://github.com/dominionenterprises/util-http-php/pulls)
- * [Issues](https://github.com/dominionenterprises/util-http-php/issues)
+ * [Pull Requests](https://github.com/traderinteractive/util-http-php/pulls)
+ * [Issues](https://github.com/traderinteractive/util-http-php/issues)
 
 ##Project Build
 With a checkout of the code get [Composer](http://getcomposer.org) in your PATH and run:
```
Comment (chadicus) [inline README.md]: you removed the `build.php` so the comment below is now invalid

File: src/HttpException.php [hunk 4]
```diff
@@ -1,55 +1,55 @@
...
-        $userMessage = null
+        string $message = 'Application Error',
+        int $httpStatusCode = 500,
+        int $code = 0,
+        Exception $previous = null,
+        string $userMessage = null
     ) {
         if (!is_string($message)) {
-            throw new \InvalidArgumentException('$message was not a string');
+            throw new InvalidArgumentException('$message was not a string');
```
Comment (chadicus) [inline src/HttpException.php]: Since you're using type hints, these exceptions will never be thrown

File: src/Util/Http.php [hunk 5]
```diff
@@ -46,12 +44,12 @@
...
      *
-     * @throws \Exception Thrown if unable to parse the headers
+     * @throws Exception Thrown if unable to parse the headers
      */
-    public static function parseHeaders($rawHeaders)
+    public static function parseHeaders(string $rawHeaders) : array
     {
         if (!is_string($rawHeaders) || trim($rawHeaders) === '') {
-            throw new \InvalidArgumentException('$rawHeaders was not a string');
+            throw new InvalidArgumentException('$rawHeaders was not a string');
```
Comment (chadicus) [inline src/Util/Http.php]: with type hints, the `is_string` check is not needed.

File: src/Util/Http.php [hunk 6]
```diff
@@ -82,23 +80,35 @@ public static function parseHeaders($rawHeaders)
...
             }
 
-            throw new \Exception("Unsupported header format: {$field}");
+            throw new Exception("Unsupported header format: {$field}");
         }
 
         return $headers;
     }
 
+    private static function setRequest($match, array $headers) : array
```
Comment (chadicus) [inline src/Util/Http.php]: type hint match

File: src/Util/Http.php [hunk 7]
```diff
@@ -82,23 +80,35 @@ public static function parseHeaders($rawHeaders)
...
+    {
+        $headers['Request Method'] = trim($match[1]);
+        $headers['Request Url'] = trim($match[2]);
+        return $headers;
+    }
+
+    private static function setResponse($match, array $headers) : array
```
Comment (chadicus) [inline src/Util/Http.php]: type hint match

**Review by chadicus — CHANGES_REQUESTED** [review]
Looks like you forgot to change the namespace.
There is also a few issues with exceptions (see comments)
Also look at the [README.md](https://github.com/hotline-emu/util-http-php/blob/imp/de-trader/README.md) in a browser. Makes sure everything makes sense.

File: src/HttpException.php [hunk 8]
```diff
@@ -1,55 +1,43 @@
...
-     * @param \Exception $previous @see \Exception::__construct()
+     * @param int $code @see Exception::__construct()
+     * @param Exception $previous @see Exception::__construct()
      * @param string|null $userMessage a nicer message to display to the user sans sensitive details
      *
-     * @throws \InvalidArgumentException if $message is not a string
-     * @throws \InvalidArgumentException if $httpStatusCode is not an int
-     * @throws \InvalidArgumentException if $code is not an int
-     * @throws \InvalidArgumentException if $userMmessage is not null and is not a string
+     * @throws InvalidArgumentException if $message is not a string
```
Comment (chadicus) [inline src/HttpException.php]: Most of these are no longer thrown

File: src/HttpException.php [hunk 9]
```diff
@@ -1,55 +1,43 @@
...
-
-        if (!is_int($httpStatusCode)) {
-            throw new \InvalidArgumentException('$httpStatusCode was not an int');
-        }
-
-        if (!is_int($code)) {
-            throw new \InvalidArgumentException('$code was not an int');
-        }
-
         if ($userMessage !== null && !is_string($userMessage)) {
```
Comment (chadicus) [inline src/HttpException.php]: this will never be true, b/c of type hinting

File: src/Util/Http.php [hunk 10]
```diff
@@ -82,23 +76,35 @@ public static function parseHeaders($rawHeaders)
...
             }
 
-            throw new \Exception("Unsupported header format: {$field}");
+            throw new Exception("Unsupported header format: {$field}");
         }
 
         return $headers;
     }
 
+    private static function setRequest(array $match, array $headers) : array
```
Comment (chadicus) [inline src/Util/Http.php]: consider a better name for both `setRequest` and `setResponse`.

File: src/Util/Http.php [hunk 11]
```diff
@@ -213,13 +219,13 @@ public static function getQueryParams($url, array $collapsedParams = [])
...
      *
-     * @throws \InvalidArgumentException if $url was not a string
-     * @throws \Exception if a parameter is given as array but not included in the expected array argument
+     * @throws InvalidArgumentException if $url was not a string
+     * @throws Exception if a parameter is given as array but not included in the expected array argument
      */
-    public static function getQueryParamsCollapsed($url, array $expectedArrayParams = [])
+    public static function getQueryParamsCollapsed(string $url, array $expectedArrayParams = []) : array
     {
         if (!is_string($url)) {
```
Comment (chadicus) [inline src/Util/Http.php]: This will never be true due to type hinting

**Review by chadicus — CHANGES_REQUESTED** [review]

File: composer.json [hunk 12]
```diff
@@ -20,13 +20,16 @@
...
     "require-dev": {
-        "phpunit/phpunit": "~5.0",
-        "satooshi/php-coveralls": "~0.6.1",
-        "squizlabs/php_codesniffer": "~2.0"
+        "php-coveralls/php-coveralls": "^1.0",
+        "phpunit/phpunit": "^6.0",
+        "squizlabs/php_codesniffer": "^3.2"
     },
     "autoload": {
         "psr-4": { "DominionEnterprises\\": "src" }
```
Comment (chadicus) [inline composer.json]: Need to update the namespace

File: src/HttpException.php [hunk 13]
```diff
@@ -1,57 +1,35 @@
 <?php
-/**
- * Defines the \DominionEnterprises\HttpException class.
- */
 
 namespace DominionEnterprises;
```
Comment (chadicus) [inline src/HttpException.php]: namespace

File: src/Util/Http.php [hunk 14]
```diff
@@ -1,11 +1,8 @@
 <?php
-/**
- * Defines the \DominionEnterprises\Util\Http class.
- */
 
 namespace DominionEnterprises\Util;
```
Comment (chadicus) [inline src/Util/Http.php]: namespace

File: src/Util/Http.php [hunk 15]
```diff
@@ -46,14 +43,10 @@
      *
      * @return array the parsed headers
      *
-     * @throws \Exception Thrown if unable to parse the headers
+     * @throws Exception Thrown if unable to parse the headers
      */
-    public static function parseHeaders($rawHeaders)
+    public static function parseHeaders(string $rawHeaders) : array
     {
-        if (!is_string($rawHeaders) || trim($rawHeaders) === '') {
```
Comment (chadicus) [inline src/Util/Http.php:53]: you should probably keep the `trim($rawheaders)` check

**Review by chadicus — CHANGES_REQUESTED** [review]
You still need to update the namespace of the library

File: src/Util/Http.php (hunk 15 above)
Comment (hotline-emu) [reply src/Util/Http.php:53]: done

**Review by chadicus — APPROVED** [review]

**Pull request merged**